src/shared/useComputePositionId.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState, useRef } from 'react';
+import {
+  Position,
+  PositionId,
+} from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+
+export function useComputePositionId() {
+  const ref = useRef<(position: Position) => PositionId>();
+  const [, setIsSet] = useState(false);
+
+  useEffect(() => {
+    const set = async () => {
+      // cant import directly without breaking the build cmd
+      const { computePositionId } = await import('@penumbra-zone/wasm/dex');
```

Comment:
suggestion: given this is an sync thing, can we move this into react-query?

Reply:
Done